File: java/log-levels/src/main/java/LogLevels.java
import java.util.Arrays;
import java.util.stream.Collectors;

public class LogLevels {
    
    public static String message(String logLine) {
        String[] words = logLine.split("\s");
        String result = Arrays.stream(words).skip(1).collect(Collectors.joining(" "));
        return result.trim();
    }

    public static String logLevel(String logLine) {
        StringBuilder result = new StringBuilder();
        for (int i = 0; logLine.charAt(i) != ']'; i++) {
            if ((logLine.charAt(i) != '[') && Character.isAlphabetic(logLine.charAt(i))) {
                result.append(logLine.charAt(i));
            }
        }
        return result.toString().toLowerCase();
    }

    public static String reformat(String logLine) {
        return message(logLine) + ' ' + '(' + logLevel(logLine) + ')';
    }
}

File: solutions/cpp/log-levels/1/log_levels.cpp
#include <string>

namespace log_line {
std::string message(std::string line) {
    return line.substr(line.find(" ") + 1);
}

std::string log_level(std::string line) {
    size_t l{line.find("[") + 1};
    size_t r{line.find("]") - 1};
    return line.substr(l, r);
}

std::string reformat(std::string line) {
    return log_line::message(line) + " (" + log_line::log_level(line) + ")";
}
} // namespace log_line
